### DISCUSSION TITLE
Troubleshooting cors-proxy HTTP 500

### DISCUSSION THREAD

**Discussion by tmeuze** [Lissy93/dashy#827] [Help and Support]
Hi there, 

I am having the hardest time with certain widgets - specifically of self-hosted services - and cors-proxy. In the logs, I'm getting a very generic HTTP 500 error on client-side, and no logs for the container.

![CleanShot 2022-07-31 at 17 26 51](https://user-images.githubusercontent.com/9298659/182046595-07d204e5-bcb9-41cd-808d-0081b26dbd8c.png)

Aside the 500 error, I'm hit with endless timeout errors. I feel that might just be a side-effect of the issue. 

![CleanShot 2022-07-31 at 18 15 22](https://user-images.githubusercontent.com/9298659/182047420-58189bce-6812-4fd2-ab19-f7b126cecb3d.png)

I am using Traefik. I have CORS middleware set up as such:
```
      headers:
        accessControlAllowOriginList: 
          - "*"
```
CORS Proxy is on by default:
`widgetsAlwaysUseProxy: true`

It should be noted that I'm using an "internal-only" Traefik instance; the FQDN is not publicly addressable. I use a self-signed TLS certificate that clients blindly trust. However, I briefly routed a public FQDN to it, and also tried via direct IP, to no avail. Internet APIs like Weather, News, and Public IP work, while self-hosted service widgets like Healthchecks, Glances, and AdGuard - no matter if through Traefik or direct IP. 

Any advice would be greatly appreciated. Thank you.

**Comment by lissy93** [comment]
Usually just setting `usePoxy: true` fixes all issues,  see [Widget Docs --> Proxying Requests](https://github.com/Lissy93/dashy/blob/master/docs/widgets.md#proxying-requests) for more info. 

If you're still having issues, check the troubleshooting docs, specifically [Fixing Widget CORS Errors](https://github.com/Lissy93/dashy/blob/master/docs/troubleshooting.md#widget-cors-errors).
